src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,362 @@
+            image={{ src: item.image, avatar: true }}
+            icon={{
+              name: 'close',
+              onClick: this.onCloseIconClick.bind(this, item),
+              onKeyDown: this.onCloseIconKeyDown.bind(this, item),
+              'aria-label': `Remove ${item.header} from selection.`,
+              'aria-hidden': false,
+              role: 'button',
+            }}
+          />
+        ))
+  }
+
+  onInputFocus = () => {
+    this.setState({ focused: true })
+  }
+
+  onInputBlur = () => {
+    this.setState({ focused: false })
+  }
+
+  onContainerClick = isOpen => {
+    !isOpen && this.inputRef.focus()
+  }
+
+  onInputKeyUpIfMultiple = event => {
+    const { inputValue, active, backspaceDelete } = this.state
+
+    switch (keyboardKey.getCode(event)) {
+      case keyboardKey.Backspace:
+        if (inputValue === '' && active.length > 0) {
+          if (!backspaceDelete) {
+            this.setState({ backspaceDelete: true })
+          } else {
+            this.removeFromActive()
+          }
+        }
+      default:
+        return
+    }
+  }
+
+  onDropdownChange = element => {
+    if (this.props.multiple) {
+      const newActive = [...this.state.active, element]
+
+      _.invoke(this.props, 'onChange', newActive)
+
+      this.setState({
+        active: newActive,
+        inputValue: '',
+        message: `${element.header} has been selected.`,
+      })
+    }
+  }
+
+  onCloseIconClick = (element, event) => this.handleCloseIconAction(element, event)
+
+  onCloseIconKeyDown = (element, event) => {
+    if (keyboardKey.getCode(event) === keyboardKey.Enter) {
+      this.handleCloseIconAction(element, event)
+    }
+  }
+
+  onInputKeyDown = (highlightedIndex, selectItemAtIndex, event) => {
+    switch (keyboardKey.getCode(event)) {
+      case keyboardKey.Tab:
+        if (highlightedIndex !== undefined) {
+          selectItemAtIndex(highlightedIndex)
+        }
+        return
+      default:
+        return
+    }
+  }
+
+  stateReducer = (state: DownshiftState<any>, changes: StateChangeOptions<any>) => {
+    switch (changes.type) {
+      case Downshift.stateChangeTypes.changeInput:
+        this.setState({
+          inputValue: changes.inputValue,
+          backspaceDelete: !(state.inputValue.length > 0 && changes.inputValue.length === 0),
+        })
+        return changes
+      default:
+        return changes
+    }
+  }
+
+  private handleCloseIconAction(element, event) {
+    this.removeFromActive(element)
+    this.inputRef.focus()
+    event.stopPropagation()
+  }
+
+  private removeFromActive(element?) {
+    let { active } = this.state
+    let poppedElement = element
+
+    if (poppedElement) {
+      active = active.filter(currentElement => currentElement !== element)
+    } else {
+      poppedElement = active.pop()
+    }
+
+    _.invoke(this.props, 'onChange', active)
+
+    this.setState({
+      active,
+      message: `${poppedElement.header} has been removed.`,
```

Comment:
as we will need to be able to translate this text, we should probably pass a callback and evaluate it here, something like
```
message: props.translateRemovedMessage ? props.translateRemovedMessage(element.header) : `${element.header} has been removed.`
```